src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -526,6 +547,13 @@ private void EnsureNuGetAndEnvDTEProjectCache()
                     var supportedProjects = EnvDTESolutionUtility.GetAllEnvDTEProjects(_dte)
                         .Where(project => EnvDTEProjectUtility.IsSupported(project));
 
+                    if (!supportedProjects.Any())
+                    {
+                        ActivityLog.LogWarning(
+                                 ExceptionHelper.LogEntrySource,
+                                 $"There are no supported projects in the solution.");
```

Comment:
As this warning won't have an exception stack consider adding more context to the message to be able to associate it with code later, say
<pre>
[VsSolutionManager::EnsureNuGetAndEnvDTEProjectCache] There are no supported projects in the solution.
</pre>

You can use `nameof(EnsureNuGetAndEnvDTEProjectCache)` to get the method name.